### ISSUE TITLE
Make "default" value in option definition lazily evaluated

### ISSUE THREAD

**Issue by Rogach** [scallop/scallop#38]
subj.

**Comment by Rogach** [comment]
I implemented the feature, and published a snapshot:

``` scala
resolvers += "sonatype snapshots" at "https://oss.sonatype.org/content/repositories/snapshots"
libraryDependencies += "org.rogach" %% "scallop" % "0.7.0~1~SNAPSHOT"
```

**Comment by Rogach** [comment]
Published version `0.7.1`. It will be synced to Maven Central in 1-2 hours.
